src/m365/aad/commands/administrativeunit/administrativeunit-roleassignment-add.spec.ts

```diff
@@ -0,0 +1,292 @@
+
+  it('fails validation if administrative unit id is not a valid GUID', async () => {
+    const actual = await command.validate({
+      options: {
+        administrativeUnitId: '123',
+        roleDefinitionId: roleDefinitionId,
+        userId: userId
+      }
+    }, commandInfo);
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('fails validation if role definition id is not a valid GUID', async () => {
+    const actual = await command.validate({
+      options: {
+        administrativeUnitId: administrativeUnitId,
+        roleDefinitionId: '123',
+        userId: userId
+      }
+    }, commandInfo);
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('fails validation if user id is not a valid GUID', async () => {
+    const actual = await command.validate({
+      options: {
+        administrativeUnitId: administrativeUnitId,
+        roleDefinitionId: roleDefinitionId,
+        userId: '123'
+      }
+    }, commandInfo);
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('correctly assign a role specified by id to and administrative unit specified by id and to a user specified by id', async () => {
+    sinon.stub(roleAssignment, 'createEntraIDRoleAssignmentWithAdministrativeUnitScope').withArgs(roleDefinitionId, userId, administrativeUnitId).resolves(unifiedRoleAssignment);
+
+    await command.action(logger, {
+      options: {
+        administrativeUnitId: administrativeUnitId,
+        roleDefinitionId: roleDefinitionId,
+        userId: userId
+      }
+    });
+    //assert.strictEqual(loggerLogSpy.lastCall,'');
+    assert(loggerLogSpy.calledOnceWithExactly(unifiedRoleAssignment));
+  });
+
+  it('correctly assign a role specified by name to and administrative unit specified by name and to a user specified by name', async () => {
+    sinon.stub(aadAdministrativeUnit, 'getAdministrativeUnitByDisplayName').withArgs(administrativeUnitName).resolves({ id: administrativeUnitId, displayName: administrativeUnitName });
+    sinon.stub(aadUser, 'getUserIdByUpn').withArgs(userName).resolves(userId);
+    sinon.stub(roleDefinition, 'getDirectoryRoleDefinitionByDisplayName').withArgs(roleDefinitionName).resolves({ id: roleDefinitionId, displayName: roleDefinitionName });
+    sinon.stub(roleAssignment, 'createEntraIDRoleAssignmentWithAdministrativeUnitScope').withArgs(roleDefinitionId, userId, administrativeUnitId).resolves(unifiedRoleAssignment);
+
+    await command.action(logger, {
+      options: {
+        administrativeUnitName: administrativeUnitName,
+        roleDefinitionName: roleDefinitionName,
+        userName: userName
+      }
+    });
+    assert(loggerLogSpy.calledOnceWithExactly(unifiedRoleAssignment));
+  });
+
+  it('correctly handles error', async () => {
+    sinon.stub(roleAssignment, 'createEntraIDRoleAssignmentWithAdministrativeUnitScope').throws(Error('Invalid request'));
+
+    await assert.rejects(command.action(logger, { options: {} } as any), new CommandError('Invalid request'));
```

Comment:
Let's pass valid value into the command, otherwise it's hard to tell what we're testing exactly.